util/topgen/merge.py

```diff
@@ -41,7 +41,7 @@ def amend_ip(top, ip):
     # Size
     if not "size" in ip_module:
         ip_module["size"] = "0x%x" % max(ip["gensize"], 0x1000)
-    elif ip_module["size"] < ip["gensize"]:
+    elif ip_module["size"] < "0x%x" % ip["gensize"]:
```

Comment:
You cannot really compare strings in this way, but you can compare integers. Since size and gensize are coming from hjson where these numbers might be strings, the quick fix might be:

```suggestion
    elif int(ip_module["size"], 0) < ip["gensize"]:
```

(The real fix would probably be to parse `size` to an int from very early on, but that's for another day.)

Reply:
@imphil thank you for reviewing and suggesting a better fix. I just updated accordingly.